.github/workflows/main.yml

```diff
@@ -149,7 +149,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python-version: [3.8, 3.9, '3.10', 3.11, 3.12]
+        python-version: [3.8, 3.9, 3.10, 3.11, 3.12]
```

Comment:
As you can see from the failure the `s around 3.10 were intentional otherwise its interpreted as 3.1 something none of the other numbers have that issue with, While I guess they all could have been quoted it was only needed for 3.10.